iD tests fail waiting for preset and other data

As of OpenHistoricalMap/iD@148c32ac3c2bde08e58fddec74900d7be170a4fa, running `npm test` produces 28 test failures that appear to be caused by a failure to load data that would normally be fetched over the network. This might be related to #720, though simply mocking the response for that request doesn’t address these failures:

```
Chrome Headless 125.0.6368.2 (Mac OS 10.15.7) iD.presetIndex #match returns a collection containing presets matching a geometry and tags FAILED
	Error: Timeout of 5000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.
Chrome Headless 125.0.6368.2 (Mac OS 10.15.7): Executed 1414 of 2083 (1 FAILED) (skipped 11) (0 secs / 6.428 secs)
Chrome Headless 125.0.6368.2 (Mac OS 10.15.7) iD.presetIndex #match returns a collection containing presets matching a geometry and tags FAILED
Chrome Headless 125.0.6368.2 (Mac OS 10.15.7) iD.presetIndex #match returns the appropriate fallback preset when no tags match FAILED
	Error: Timeout of 5000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.
Chrome Headless 125.0.6368.2 (Mac OS 10.15.7): Executed 1415 of 2083 (2 FAILED) (skipped 11) (0 secs / 11.43 secs)
Chrome Headless 125.0.6368.2 (Mac OS 10.15.7) iD.presetIndex #match returns the appropriate fallback preset when no tags match FAILED
Chrome Headless 125.0.6368.2 (Mac OS 10.15.7) iD.presetIndex #match matches vertices on a line as points FAILED
	Error: Timeout of 5000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.
```

```
Chrome Headless 125.0.6368.2 (Mac OS 10.15.7) iD.serviceOsmWikibase creates correct sitelinks FAILED
	AssertionError: expected 'Key:amenity' to deeply equal 'Key: amenity'
	    at Context.<anonymous> (test/spec/services/osm_wikibase.js:307:49)
Chrome Headless 125.0.6368.2 (Mac OS 10.15.7): Executed 1635 of 2083 (19 FAILED) (skipped 18) (0 secs / 1 min 34.502 secs)
Chrome Headless 125.0.6368.2 (Mac OS 10.15.7) iD.serviceOsmWikibase creates correct sitelinks FAILED
	AssertionError: expected 'Key:amenity' to deeply equal 'Key: amenity'
Chrome Headless 125.0.6368.2 (Mac OS 10.15.7) iD.serviceOsmWikibase #getEntity calls the given callback with the results of the getEntity data item query FAILED
	Error: Uncaught AssertionError: expected { action: 'wbgetentities', …(6) } to deeply equal { action: 'wbgetentities', …(6) } (node_modules/chai/chai.js:250)
Chrome Headless 125.0.6368.2 (Mac OS 10.15.7): Executed 1638 of 2083 (20 FAILED) (skipped 18) (0 secs / 1 min 34.555 secs)
```

```
Chrome Headless 125.0.6368.2 (Mac OS 10.15.7) iD.validations.suspicious_name ignores feature with a specific name FAILED
	Error: Uncaught TypeError: Cannot read properties of undefined (reading 'get') (dist/iD.js:75473:7 <- dist/iD.js:1637)
Chrome Headless 125.0.6368.2 (Mac OS 10.15.7): Executed 2055 of 2083 (21 FAILED) (skipped 19) (0 secs / 1 min 40.206 secs)
```

Thanks to @pedromml for reporting the issue.

--- a/modules/presets/index.js
+++ b/modules/presets/index.js
@@ -23,7 +23,9 @@ export { _mainPresetIndex as presetManager };
  * Sets preset defaults specific to OpenHistoricalMap.
  */
 function setHistoricalDefaults(defaults) {
-  defaults.relation.unshift('type/chronology');
+  if (defaults.relation) {
+    defaults.relation.unshift('type/chronology');
+  }
 }
 
 /**
@@ -45,14 +47,18 @@ function addHistoricalPresets(presets) {
  * Adds fields specific to OpenHistoricalMap.
  */
 function addHistoricalFields(fields) {
-  fields.start_date.type = 'date';
-  fields.end_date = {
-    ...fields.start_date,
-    key: 'end_date'
-  };
+  if (fields.start_date) {
+    fields.start_date.type = 'date';
+    fields.end_date = {
+      ...fields.start_date,
+      key: 'end_date'
+    };
+  }
 
   // A combo box would encourage mappers to choose one of the suggestions, but we want mappers to be as detailed as possible.
-  fields.source.type = 'text';
+  if (fields.source) {
+    fields.source.type = 'text';
+  }
 
   fields.license = {
     key: 'license',

--- a/modules/services/osm_wikibase.js
+++ b/modules/services/osm_wikibase.js
@@ -105,7 +105,7 @@ export default {
         } else {
             prefix = type + ':';
         }
-        return prefix + (value ? `${key}=${value}` : key).replace(/_/g, ' ').trim();
+        return (prefix + (value ? `${key}=${value}` : key).replace(/_/g, ' ')).trim();
     },
 
 

--- a/test/spec/services/osm_wikibase.js
+++ b/test/spec/services/osm_wikibase.js
@@ -284,7 +284,7 @@ describe('iD.serviceOsmWikibase', function () {
           {
             action: 'wbgetentities',
             sites: 'wiki',
-            titles: 'Locale:fr|Key:amenity|Tag:amenity=parking',
+            titles: 'Locale:fr|Key:amenity|OpenHistoricalMap/Tags/Key/amenity|Tag:amenity=parking|OpenHistoricalMap/Tags/Tag/amenity=parking',
             languages: 'fr',
             languagefallback: '1',
             origin: '*',

--- a/test/spec/spec_helpers.js
+++ b/test/spec/spec_helpers.js
@@ -83,6 +83,7 @@ iD.localizer.loadLocale('en', 'tagging');
 
 // Initializing `coreContext` initializes `_background`, which tries loading:
 cached.imagery = [];
+cached.wayback = {};
 // Initializing `coreContext` initializes `_presets`, which tries loading:
 cached.preset_categories = {};
 cached.preset_defaults = {};

